libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -48,13 +48,58 @@ void StaticAnalyzer::endVisit(ContractDefinition const&)
 
 bool StaticAnalyzer::visit(FunctionDefinition const& _function)
 {
+	m_localVarUseCount.clear();
 	m_nonPayablePublic = _function.isPublic() && !_function.isPayable();
 	return true;
 }
 
 void StaticAnalyzer::endVisit(FunctionDefinition const&)
 {
 	m_nonPayablePublic = false;
+	for (auto var = m_localVarUseCount.begin(); var != m_localVarUseCount.end(); ++var)
+	{
+		if (var->second == 0)
+		{
+			warning(var->first->location(), "Unused local variable");
+		}
+	}
+}
+
+bool StaticAnalyzer::visit(Identifier const& _identifier)
+{
+	if (auto var = dynamic_cast<VariableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+	{
+		if (var->isLocalVariable() && var->name() != "")
+		{
+			if (m_localVarUseCount.find(var) != m_localVarUseCount.end())
+			{
+				m_localVarUseCount[var] += 1;
+			}
+		}
+	}
+	return true;
+}
+
+void StaticAnalyzer::endVisit(Identifier const&)
+{
+}
+
+bool StaticAnalyzer::visit(VariableDeclaration const& _variable)
+{
+	// There's no real way to get a good tracking on unnamed variables in someone's
+	// code.  These can show up in the returns clause of a function declaration,
+
+	if (_variable.isLocalVariable() && _variable.name() != "")
+	{
+		// We don't have to check if the variable was already inserted
```

Comment:
The comment is clearly ambiguous.  I was attempting to say that we don't have to worry about something like:
`uint a;
uint a;`

Because the parser would have already rejected it.  I can just remove the comment.  If redundant declarations had not already been filtered out, then `m_localVarUseCount[&_variable] = 0` isn't particularly safe to use because it would cause the counter to reset to 0 when the next declaration was encountered.